proxy/client.go

```diff
@@ -29,12 +29,17 @@ func newClient(dial func() (net.Conn, error), i interceptor) *client {
 func (c *client) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	err := c.InterceptRequest(r)
 	if err == docker.ErrNoSuchImage {
-		http.NotFound(w, r)
+		http.Error(w, err.Error(), http.StatusNotFound)
 		return
 	}
 	if err != nil {
-		http.Error(w, "Unable to intercept request", http.StatusInternalServerError)
-		Warning.Print("Error intercepting request: ", err)
+		switch err.(type) {
+		case *docker.NoSuchContainer:
+			http.Error(w, err.Error(), http.StatusNotFound)
+		default:
+			http.Error(w, "Unable to intercept request", http.StatusInternalServerError)
+			Warning.Print("Error intercepting request: ", err)
+		}
```

Comment:
ErrNoSuchImage is not a type, just an `errors.New()`, so we need 2 separate checks.